File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        
        s = s.toLowerCase().replaceAll("[^a-zA-Z0-9]","");
        
        for(int i=0,j=s.length()-1;i<s.length()/2;i++,j--){
            if(i<=j && s.charAt(i) != s.charAt(j)){
                // break;
                return false;
            }
        }
        return true;    
        }
}

File: 0125-valid-palindrome/0125-valid-palindrome.ts
function isPalindrome(s: string): boolean {
    
    const input: string = s.toLowerCase().replace(/[^a-zA-Z0-9]/g, '');
    let i = 0;
    let j = input.length - 1;
    
    while (i <= j) {

        if (input.charAt(i) !== input.charAt(j)) {
            return false;
        }
        i++;
        j--;
    }

    return true;
};